### PR TITLE
[5.3] FILTER_VALIDATE_FLOAT

### PR DESCRIPTION
**Pull request by halaei**
Floats are now should be allowed.

### CODE REVIEW

File: src/Illuminate/Cache/Repository.php [hunk 1]
```diff
@@ -196,7 +196,7 @@ public function pull($key, $default = null)
      */
     public function put($key, $value, $minutes = null)
     {
-        if (is_array($key) && filter_var($value, FILTER_VALIDATE_INT) !== false) {
+        if (is_array($key) && filter_var($value, FILTER_VALIDATE_FLOAT) !== false) {
```
Comment (halaei) [inline src/Illuminate/Cache/Repository.php]: Or just remove the check. Maybe DateTime should be allowed to

**Comment by GrahamCampbell** [comment]
That is true. Could you add a couple more tests please?

**Comment by halaei** [comment]
Don't know exactly why the check was here. It seems that I can remove it. By more test, do you mean unit test or check for DateTime?

**Comment by GrahamCampbell** [comment]
No, this code looks correct.

**Comment by GrahamCampbell** [comment]
Please read it again. :)

**Comment by halaei** [comment]
You know what I mean, but I don't know what you mean. So could you add a couple of more tests yourself? Thanks.

File: src/Illuminate/Cache/Repository.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Illuminate/Cache/Repository.php]: No, the point is datetime makes it through to the next part.

**Comment by halaei** [comment]
I don't get it. These codes don't work:
```php
Cache::put(['foo'=>'bar2'], \Carbon\Carbon::now()->addMinutes(2));
Cache::put(['foo'=>'bar2'], 1.3);
echo Cache::get('foo'); // should be bar2 but is not!
```

**Comment by GrahamCampbell** [comment]
Ping @taylorotwell.

**Comment by taylorotwell** [comment]
I think you can probably just check is_array($key)... I'm not sure any other checks are needed.

**Comment by taylorotwell** [comment]
Is there any situation where you can pass an array to key **without** wanting to use `putMany`?

**Comment by halaei** [comment]
Don't know. May be we can find a reason if we check the history of code?

**Comment by taylorotwell** [comment]
There is no documented example of calling “put” with an array. I would suggest just doing the is_array check and removing the $value check entirely.

**Pull request merged**